### PR TITLE
Make selection writers more pythonic

### LINKED ISSUE #1244: NDX selection writer make surprising use of the `mode` kwarg
### Expected behaviour

In the following snippet, I expect all the groups to be written in the index file:

```python
with mda.selections.gromacs.SelectionWriter('leaflets.ndx', mode='w') as ndx:
    ndx.write(upper_saturated, name='upper_sat')
    ndx.write(upper_saturated, name='lower_sat')
    ndx.write(upper_saturated, name='upper_unsat')
    ndx.write(upper_saturated, name='lower_unsat')
```

The `mode` keyword argument refers to the mode the file should be open in. I want a new file to be created so I can write on it, so I set the mode to `w` (`wa` by default, but I want a new file!).

### Actual behaviour

Only the last group is written in the resulting file. It looks like every call to write open the file in the mode selected above, write, and then close the file. This is not consistent with how files are handled in python, especially within a context manager.

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
'0.16.0-dev0'

### PR DESCRIPTION
**Pull request by jbarnoud**
* Remove the `wa` mode
* Add a `close` method
* Close the file when leaving the context manager rather than at every
  write
* Factorize the tests

Fixes #1244

PR Checklist
------------
 - [X] Tests?
 - [X] Docs?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/selections/base.py [hunk 1]
```diff
@@ -105,15 +110,15 @@ class SelectionWriterBase(six.with_metaclass(_Selectionmeta)):
...
     default_numterms = 8
 
-    def __init__(self, filename, mode="wa", numterms=None, preamble=None, **kwargs):
+    def __init__(self, filename, mode="w", numterms=None, preamble=None, **kwargs):
         """Set up for writing to *filename*.
 
         :Arguments:
            *filename*
                output file
            *mode*
```
Comment (kain88-de) [inline package/MDAnalysis/selections/base.py]: while you are at it. Can you also update the docs to the numpy format. Thanks

Comment (jbarnoud) [inline package/MDAnalysis/selections/base.py]: Done.

**Comment by orbeckst** [comment]
Regarding docs: I think you can ignore my comment and we just add it as a TODO to #1104 .

**Comment by orbeckst** [comment]
I will squash & merge later today unless there are other opinions.

**Pull request merged**